### PR TITLE
Add some more commands.

### PR DESCRIPTION
**Pull request by tomprince**

### CODE REVIEW

**Comment by tomprince** [comment]
The failure is due to https://github.com/Manishearth/rust-clippy/issues/1476

File: src/client.rs [hunk 1]
```diff
@@ -455,6 +455,27 @@ impl<S: Read + Write> Client<S> {
...
+    /// Lists unique tags values of the specified type for songs matching the given query.
+    // TODO: list type [filtertype] [filterwhat] [...] [group] [grouptype] [...]
+    // It isn't clear if or how `group` works
+    pub fn list(&mut self, term: &Term, query: &Query) -> Result<Vec<String>> {
+        self.run_command("list", (term, query))
+            .and_then(|_| self.read_pairs().map(|p| p.map(|p| p.1)).collect())
+    }
+
+    /// Find all songs in the db that match query and adds them to current playlist.
+    pub fn findadd(&mut self, query: &Query) -> Result<()> {
```
Comment (kstep) [inline src/client.rs]: I thought of joining up `searchadd` and `findadd` together into a single method, as the functionality of these commands are essentially the same, except for case-sensitivity, which can be passed either as a separate parameter, or as a field in `Query` object (debatable).

Comment (tomprince) [inline src/client.rs]: On one hand, having something like that would make sense. On the other, keeping them separate makes it easier for somebody looking at MPD documentation to figure out what the corresponding method is. Given that most of the methods currently do match with the underlying MPD commands, I'd be inclined to at least have a lower-level interface that more closely mirrors the underlying protocol.

Comment (kstep) [inline src/client.rs]: I'll merge it as is for now, but I think this point should be thought through more thoroughly.

**Pull request merged**